pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -241,7 +241,9 @@ func newLibvirtDomainManager(connection cli.Connection, virtShareDir, ephemeralD
 		}
 
 		if len(list) == 0 {
-			return nil, nil
+			// We must return an error, otherwise the nil value will be cached
+			// and will make the scraper to fail
+			return nil, errors.New("empty DomainStats")
```

Comment:
Can you add a test for it? If we have a test we should also be able to drop the comment.

Reply:
Yes, I can add a test (and I will :) ), but I still think the comment should stay. Mostly because the cache implementation IMO is incorrect, so the comment can serve as a warning. Because, this test will not prevent, in a different part of the code, passing a func callback that returns `nil` to `virtcache.NewTimeDefinedCache()`.

The cache probably can be changed to store `*T` instead of `T`, and check for `nil` to avoid caching it. I didn't touch it, because I think it should be part of libvirt (if it really is necessary, that I'm not sure). Also, I have few issues with its API, that should be fixed in a different PR, in case we want to keep it.

Reply:
Done